cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/ExpressionBuilder.kt

```diff
@@ -68,7 +69,17 @@ fun MetadataProvider.newBinaryOperator(
     code: String? = null,
     rawNode: Any? = null
 ): BinaryOperator {
-    val node = BinaryOperator()
+    val node =
+        if (
+            this is LanguageProvider &&
+                (this.language as? HasShortCircuitOperators)
+                    ?.operatorCodes
+                    ?.contains(operatorCode) == true
+        ) {
+            ShortCircuitOperator()
```

Comment:
We should probably document this magic in the documentation of this node builder, since it is a little bit different than the others.